### PR TITLE
NIFI-3355 Allows NiFi to bind to specific network interfaces, with se…

### PR DESCRIPTION
**Pull request by jtstorck**
…parate interface lists for HTTP and HTTPS.

### CODE REVIEW

File: nifi-nar-bundles/nifi-framework-bundle/nifi-framework/nifi-web/nifi-jetty/src/main/java/org/apache/nifi/web/server/JettyServer.java [hunk 1]
```diff
@@ -569,17 +572,43 @@ private void configureConnectors(final Server server) throws ServerConfiguration
...
 
-            // create the connector
-            final ServerConnector http = new ServerConnector(server, new HttpConnectionFactory(httpConfiguration));
+            final List<Connector> serverConnectors = Lists.newArrayList();
 
-            // set host and port
-            if (StringUtils.isNotBlank(props.getProperty(NiFiProperties.WEB_HTTP_HOST))) {
-                http.setHost(props.getProperty(NiFiProperties.WEB_HTTP_HOST));
+            final Map<String, String> httpNetworkInterfaces = props.getHttpNetworkInterfaces();
+            if (httpNetworkInterfaces.isEmpty()) {
```
Comment (jtstorck) [inline nifi-nar-bundles/nifi-framework-bundle/nifi-framework/nifi-web/nifi-jetty/src/main/java/org/apache/nifi/web/server/JettyServer.java]: This needs to be expanded to check that the map is empty, or that there are no actual values for any of the keys in the map.  As it stands, if the properties for binding to a NIC are absent from the config, this code works...  But if a property exists but is left blank, this if block would get skipped, it'd go into the else block, and end up not creating any connectors.

File: nifi-nar-bundles/nifi-framework-bundle/nifi-framework/nifi-web/nifi-jetty/src/main/java/org/apache/nifi/web/server/JettyServer.java [hunk 2]
```diff
@@ -590,28 +619,59 @@ private void configureConnectors(final Server server) throws ServerConfiguration
...
-            httpsConfiguration.setSecurePort(props.getSslPort());
-            httpsConfiguration.addCustomizer(new SecureRequestCustomizer());
+            final List<Connector> serverConnectors = Lists.newArrayList();
 
-            // build the connector
-            final ServerConnector https = new ServerConnector(server,
-                    new SslConnectionFactory(createSslContextFactory(), "http/1.1"),
-                    new HttpConnectionFactory(httpsConfiguration));
+            final Map<String, String> httpsNetworkInterfaces = props.getHttpsNetworkInterfaces();
+            if (httpsNetworkInterfaces.isEmpty()) {
```
Comment (jtstorck) [inline nifi-nar-bundles/nifi-framework-bundle/nifi-framework/nifi-web/nifi-jetty/src/main/java/org/apache/nifi/web/server/JettyServer.java]: The following comment from line 578 applies here as well. [https://github.com/apache/nifi/pull/1508/commits/88520369d910dff61892b86841dc9eab28bf03e3#r101147915](url)

**Comment by jtstorck** [comment]
Documentation for the new properties are needed.  Will work on getting that added to the adoc tonight.

**Comment by brosander** [comment]
Reviewing

**Comment by jtstorck** [comment]
Documentation has been updated for the new properties, and proper property detection for nifi.web.http(s).network.interface has been added.

**Comment by brosander** [comment]
+1 merging

**Pull request closed without merging**